text/0605-eks-rewrite.md

```diff
@@ -0,0 +1,476 @@
+  }),
+]);
+```
+
+## Cluster Configuration
+
+### New Feat: Create EKS Cluster in an isolated VPC
+
+To create a EKS Cluster in an isolated VPC, vpc endpoints need to be set for
+different AWS services (EC2, S3, STS, ECR and anything the service needs).
+
+```
+const vpc = new ec2.Vpc(this, 'vpc', {
+    subnetConfiguration: [
+        {
+            cidrMask: 24,
+            name: 'Private',
+            subnetType: ec2.SubnetType.PRIVATE_ISOLATED,
+        },
+    ],
+    gatewayEndpoints: {
+        S3: {
+            service: ec2.GatewayVpcEndpointAwsService.S3,
+        },
+    },
+});
+vpc.addInterfaceEndpoint('stsEndpoint', {
+    service: ec2.InterfaceVpcEndpointAwsService.STS,
+    open: true,
+});
+
+vpc.addInterfaceEndpoint('ec2Endpoint', {
+    service: ec2.InterfaceVpcEndpointAwsService.EC2,
+    open: true,
+});
+
+vpc.addInterfaceEndpoint('ecrEndpoint', {
+    service: ec2.InterfaceVpcEndpointAwsService.ECR,
+    open: true,
+});
+
+vpc.addInterfaceEndpoint('ecrDockerEndpoint', {
+    service: ec2.InterfaceVpcEndpointAwsService.ECR_DOCKER,
+    open: true,
+});
+
+const cluster = new eks.Cluster(this, 'MyMycluster123', {
+    version: eks.KubernetesVersion.V1_31,
+    authenticationMode: eks.AuthenticationMode.API,
+    vpc,
+    vpcSubnets: [{ subnetType: ec2.SubnetType.PRIVATE_ISOLATED }]
+});
+```
+
+### Logging Configuration
+
+Logging property is renamed from clusterLogging to logging since there is only
+one logging property in the construct.
+
+Before
+
+```
+const cluster = new eks.Cluster(this, 'Cluster', {
+  // ...
+  version: eks.KubernetesVersion.V1_31,
+  clusterLogging: [
+    eks.ClusterLoggingTypes.API,
+    eks.ClusterLoggingTypes.AUTHENTICATOR,
+  ],
+});
+```
+
+After
+
+```
+const cluster = new eks.Cluster(this, 'Cluster', {
+  version: eks.KubernetesVersion.V1_31,
+  logging: [
+    eks.ClusterLoggingTypes.API,
+    eks.ClusterLoggingTypes.AUTHENTICATOR,
+  ],
+});
+```
+
+### Output Configuration
+
+A new property `outputInfo` will replace the current 3 output properties.
+Although 3 separate output properties provide customization on output
+configuration, it increased the cognitive load and doesn’t provide a clean
+usage. The proposal here is to have one single flag to control all of them.
+
+Before
+
+```
+const cluster = new eks.Cluster(this, 'Cluster', {
+  version: eks.KubernetesVersion.V1_31,
+  outputMastersRoleArn: true,
+  outputClusterName: true,
+  outputConfigCommand: true,
+});
+```
+
+After
+
+```
+const cluster = new eks.Cluster(this, 'Cluster', {
+  version: eks.KubernetesVersion.V1_31,
+  outputInfo: true,
+});
+```
+
+### Kubectl Handler Configuration
+
+KubectlProvider is a lambda function that CDK deploys alongside the EKS cluster
+in order to execute kubectl commands against the cluster.
+
+A common scenarios is that users create a CDK app that deploys the EKS cluster,
+which is then imported in other apps in order to deploy resources onto the
+cluster.
+
+Difference
```

Comment:
Please extend this to show how resources can be deployed to imported clusters, not just how to import the kubectl provider. The ergonomics around this experience is very awkward and I would like to improve it. 

Right now, to deploy resources onto an imported cluster, we can do:

```ts
const clusterWithKubectlHandler = eks.Cluster.fromAttributes(this, 'Cluster', {
  kubectlFunctionArn: '',
});

clusterWithKubectlHandler.addManifest(...) // will reuse the existing kubectl function

const clusterWithoutKubectlHandler = eks.Cluster.fromAttributes(this, 'Cluster', {
});

// will create a new kubectl function that can deploy to the cluster
// but in this case, additional attributes must be provided when importing the cluster (e.g the role)
// this is confusing. 
clusterWithoutKubectlHandler.addManifest(...) 
```

Reply:
Add different scenarios for importing cluster/kubectl handler